### PR TITLE
:bento: Extra data and pre-batch shuffle on train datapipe

### PR DESCRIPTION
**Pull request by weiji14**
## What I am changing

- Adding more imagery data to the training set (temporarily commented out though, see https://github.com/developmentseed/chabud2023/pull/14#discussion_r1211767142)
- Implement proper shuffling to mix images with burned areas (1 labels) and no burned areas (0 labels)

## How I did it

- Added california_3.hdf5 and california_4.hdf5 to the datapipe from https://huggingface.co/datasets/chabud-team/chabud-extra/commit/7da36fcb240ef39beed1f877acc837b98746f35b
- Use [`.shuffle`](https://pytorch.org/data/0.6/generated/torchdata.datapipes.iter.Shuffler.html) instead of [`.in_batch_shuffle`](https://pytorch.org/data/0.6/generated/torchdata.datapipes.iter.InBatchShuffler.html)

## How you can test it

- Run `python trainer.py fit --trainer.max_epochs=30 --data.batch_size=6` locally.

## Related Issues

Note that the shuffling operation is slower than in-batch shuffling. There is a longer delay at the start as the image chips are added to the shuffle buffer, and each mini-batch is now processing about 2x slower (one iteration used to take ~1s, now it takes ~2s).

### CODE REVIEW

File: chabud/datapipe.py [hunk 1]
```diff
@@ -206,16 +210,17 @@ def setup(
...
-            num_instances=2, classifier_fn=_train_val_fold, buffer_size=2048
+            num_instances=2, classifier_fn=_train_val_fold, buffer_size=-1
         )
 
         # Step 4 - Convert from xarray.Dataset to tuple of torch.Tensor objects
-        # Also do batching, shuffling (for train set only) and tensor stacking
+        # Also do shuffling (for train set only), batching, and tensor stacking
         self.datapipe_train = (
-            dp_train.map(fn=_pre_post_mask_tuple)
+            dp_train.shuffle(buffer_size=100)
```
Comment (weiji14) [inline chabud/datapipe.py:225]: Default buffer size of 10000 was too slow (waited for minutes but the model never starts training). @srmsoumya, could you try a few other variations of this `buffer_size` and see how performant the model is?

Comment (srmsoumya) [reply chabud/datapipe.py:225]: sure, I was facing some error with buffers & set it to -1 in my experiment, I will look at other options as well

File: chabud/datapipe.py [hunk 2]
```diff
@@ -137,6 +137,8 @@ def __init__(
             "https://huggingface.co/datasets/chabud-team/chabud-extra/resolve/main/california_0.hdf5",
             "https://huggingface.co/datasets/chabud-team/chabud-extra/resolve/main/california_1.hdf5",
             "https://huggingface.co/datasets/chabud-team/chabud-extra/resolve/main/california_2.hdf5",
+            "https://huggingface.co/datasets/chabud-team/chabud-extra/resolve/main/california_3.hdf5",
+            "https://huggingface.co/datasets/chabud-team/chabud-extra/resolve/main/california_4.hdf5",
```
Comment (srmsoumya) [inline chabud/datapipe.py]: @weiji14 we can ignore the `california_*.hdf5` files for now, as the dataset is currently imbalanced. We can add them back once we implement the mixup & cutmix augmentations.

**Review by srmsoumya — APPROVED** [review]
Looks good to me, feel free to merge.

File: chabud/datapipe.py (hunk 2 above)
Comment (weiji14) [reply chabud/datapipe.py]: Ok, I'll comment those lines out for now, done at e9b7255ae8a687ab22c3bc43b0dca7bbbfbcba4c. The data imbalance can be tracked at #11 or #12.
